src/main/groovy/org/shipkit/internal/gradle/AutoUpdateConsumerPlugin.java

```diff
@@ -0,0 +1,126 @@
+package org.shipkit.internal.gradle;
+
+import org.gradle.api.Action;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.tasks.Exec;
+import org.shipkit.internal.gradle.autoupdate.ReplaceVersionTask;
+import org.shipkit.internal.gradle.configuration.LazyConfiguration;
+import org.shipkit.internal.gradle.util.TaskMaker;
+
+public class AutoUpdateConsumerPlugin implements Plugin<Project> {
+
+    public static final String AUTO_UPDATE_CHECKOUT_BRANCH = "autoUpdateCheckoutBranch";
+    public static final String AUTO_UPDATE_REPLACE_VERSION = "autoUpdateReplaceVersion";
+    public static final String AUTO_UPDATE_GIT_COMMIT = "autoUpdateGitCommit";
+    public static final String AUTO_UPDATE_GIT_PUSH = "autoUpdateGitPush";
+
+    public static final String SHIPKIT_NEW_VERSION = "shipkitNewVersion";
+    public static final String SHIPKIT_DEPENDENCY_FILE = "shipkitDependencyFile";
+    public static final String SHIPKIT_DEPENDENCY_PATTERN = "shipkitDependencyPattern";
+    public static final String DEPENDENCY_PATTERN_DEFAULT = "org.shipkit:shipkit:{VERSION}";
+    public static final String DEPENDENCY_FILE_DEFAULT = "build.gradle";
```

Comment:
"dependency file" name is somewhat confusing - I think this is simply "build file" :)